control/control.go

```diff
@@ -1016,6 +1016,54 @@ func (p *pluginControl) CollectMetrics(id string, allTags map[string]map[string]
 	return
 }
 
+func (p *pluginControl) StreamMetrics(
+	id string,
+	allTags map[string]map[string]string,
+	maxCollectDuration time.Duration,
+	maxMetricsBuffer int64) (chan []core.Metric, chan error, []error) {
+	if !p.Started {
+		return nil, nil, []error{ErrControllerNotStarted}
+	}
+	errs := make([]error, 0)
+	pluginToMetricMap, serrs, err := p.subscriptionGroups.Get(id)
+	if err != nil {
+		controlLogger.WithFields(log.Fields{
+			"_block":                "StreamMetrics",
+			"subscription-group-id": id,
+		}).Error(err)
+		errs = append(errs, err)
+		return nil, nil, errs
+	}
+
+	if serrs != nil {
+		for _, e := range serrs {
+			errs = append(errs, e)
+		}
+	}
+	if len(pluginToMetricMap) > 1 {
```

Comment:
That is a change that has side effects. Are you trying to make the argument that the serrs returned from the subscriptionGroups.Get() call aren't needed? I'd be interested to hear the reasoning behind it b/c I'm not sure myself.